hta/utils/utils.py

```diff
@@ -174,3 +178,21 @@ def flatten_column_names(df: pd.DataFrame) -> None:
     """Flatten a DataFrame's a multiple index column names to a single string"""
     if isinstance(df.columns, pd.MultiIndex):
         df.columns = ["_".join(col).rstrip("_") for col in df.columns]
+
+
+def get_mp_pool_size(obj: object, num_objs: int) -> int:
+    """
+    Estimate the maximum pool size for multiprocessing
+
+    Args:
+        obj (object): one instance of the objects to be processed
+        num_objs (int): the total number of objects to be processed
+
+    Return:
+        the recommend pool size
+    """
+    obj_mem = get_memory_usage(obj)
+    free_mem = psutil.virtual_memory().available
+    # Leave 20% buffer for system and other processes
+    max_np = int(0.8 * free_mem / obj_mem)
+    return min(max_np, num_objs, MAX_NUM_PROCESSES, mp.cpu_count())
```

Comment:
what is the rationale behind using "MAX_NUM_PROCESSES"? the default value is 32 and it seems pretty arbitrary. `max_np` seems to a much more reasonable estimate, although it is computationally expensive. 

A faster implementation would be `min(num_objs, mp.cpu_count())`.

Reply:
The initial idea of using MAX_NUM_PROCESSES is to expose a way for user to specified num of process. But I agree with that min(max_np, num_objs, mp.cpu_count()) is good enough. 

As for why max_np is used is to prevent OOM when using mp.

Reply:
awesome!